sky/backends/backend_utils.py

```diff
@@ -2598,6 +2598,7 @@ def check_public_cloud_enabled():
     Exceptions:
         exceptions.NoCloudAccessError: if no public cloud is enabled.
     """
+    sky_check.check(quiet=True)
```

Comment:
Running `sky check` on every invocation can be expensive and would be a major performance regression:

Current master:
```
time sky launch -c test --cloud gcp --dryrun -y 
#6.1s
```

This branch:
```
time sky launch -c test --cloud gcp --dryrun -y 
# 26.5s
```

Is there some other fix possible here? 

Alternatively, can we update `sky check` to add a new arg `cloud`, which specifies which cloud to "check"? Then we can run `sky check` only on k8s.

Reply:
I think allowing `sky_check.check` to specify a cloud to check is a much better way. If we wan't to pinpoint only k8s to run the sky check for this, I'm wondering if it's better to place `sky_check.check` at the beginning of `cli.py/_launch_with_confirm`?  There are various locations calling `global_user_state.get_enabled_clouds()`, which can possibly be affected due to external deletion.